worker-core/src/main/java/com/hpe/caf/worker/core/WorkerCore.java

```diff
@@ -156,8 +158,14 @@ private void registerNewTaskImpl(final TaskInformation taskInformation, final by
             throws InvalidTaskException, TaskRejectedException
         {
             try {
-                TaskMessage tm = codec.deserialise(taskMessage, TaskMessage.class, DecodeMethod.LENIENT);
-
+                final QueueTaskMessage queueTaskMessage = codec.deserialise(taskMessage, QueueTaskMessage.class, DecodeMethod.LENIENT);
+                final TaskMessage tm = TaskMessage.from(queueTaskMessage);
+    
+                if(queueTaskMessage.getTaskData() instanceof String) {
+                    tm.setTaskData(Base64.decodeBase64((String)queueTaskMessage.getTaskData()));
+                } else {
+                    tm.setTaskData(codec.serialise(queueTaskMessage.getTaskData()));
+                }
```

Comment:
Could this logic be moved into the creation method so that we're not constructing and then immediately changing?